File: 0371-sum-of-two-integers/0371-sum-of-two-integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        check=0xFFFFFFFFFF
        MAX_INT = 0x7FFFFFFF 
        while b!=0:
            carry=(a&b)&check
            a=(a^b)&check
            b=(carry<<1)&check
        return a if a<MAX_INT  else ~(a^check)
        

File: 0371-sum-of-two-integers/0371-sum-of-two-integers.cpp
class Solution {
public:
    int getSum(int a, int b) {
        int check=0xFFFFFFFFF;
        while (b!=0){
            int carry=(a&b)&check;
            a=(a^b)&check;
            b=(carry<<1)&check;
        }
        return a<=INT_MAX?a:~(a ^ check);
        
    }
};